src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -205,10 +206,16 @@ public static ConnectorParameterType ToFormulaType(this OpenApiSchema schema, St
                     {
                         case null:
                         case "uuid":
-                        case "mquery": // For SQL Server Power Query language
+                        case "mquery": // For SQL Server Power Query language
+                        case "email":
+                        case "uri":
+                        case "byte": // Base64 encoded string
+                        case "html":
                             return new ConnectorParameterType(schema, FormulaType.String);
-
-                        case "date-time":
+
+                        case "date":
+                        case "date-time":
+                        case "date-no-tz":
                             // Consider this as a string for now
                             // $$$ Should be DateTime type
```

Comment:
>// $$$ Should be DateTime type [](http://example.com/codeflow?start=28&length=30)

Will this be a compat issue with power apps?

Reply:
This is converting from Swagger (which has dates) to Power Fx (which has dates).

For example, can we test this:
Office365Users.MyProfileV2().birthday > Date(2001, 3,1)

Reply:
Swagger/OpenAPI has dates but they are stored as strings
See [https://swagger.io/specification/v2/](https://swagger.io/specification/v2/)

The only test is around loading the swagger which is our first goal here (what Cole and Justin have reported).

Supporting dates in APIs can be done in a separated PR.

Reply:
This change will suppress exceptions during the parse phase, but the goal here is to be able to successfully use the O365 Users connector, ie, the methods are imported and work same as in PA  ...   

So if we have to add new cases for parser, that's a signal we should verify that new functionality also works during eval.

Reply:
While I agree we can warn them that 'date' and 'date-no-tz' might not work as expected as they'll be treated as strings, there is nothing new with 'date-time' which is supported as a string for a long time and without any partner complaint - and many connectors (as I remember) use 'date-time'.

I'm in favor in not delaying this PR anymore, allow Cards to load the O365 Outlook connector (not the Users one), and continue improving our code in future PRs.
(this PR isn't about supporting date formats)

Reply:
Have checked how this is managed in PA Client
In src\AppMagic\dll\AuthoringCore\Importers\DataDescription\DataDescriptionParser.cs, line 653-654 (in ParseDataType)

Their implementation is trickier as we will need to support the different formats (Iso8601ExtendedUtc, Iso8601Datetime) and different behaviors (UserLocal, NoTimeZone) - at minimum (we can easily imagine a 3rd behavior 'WithDefinedTimeZone'.

Clearly out of scope for this PR to my eyes.

Reply:
Will add comments with these findings

Reply:
Logged this issue for dates: [https://github.com/microsoft/Power-Fx/issues/1557](https://github.com/microsoft/Power-Fx/issues/1557)